src/scrapers/otsar-hachayal.js

```diff
@@ -0,0 +1,222 @@
+      const changedTransaction = txns.pop();
+      if (classList.includes('reference')) {
+        changedTransaction.description = innerText;
+      } else if (classList.includes('details')) {
+        changedTransaction.reference = innerText;
+      } else if (classList.includes('credit')) {
+        changedTransaction.credit = innerText;
+      } else if (classList.includes('debit')) {
+        changedTransaction.debit = innerText;
+      } else if (classList.includes('balance')) {
+        changedTransaction.balance = innerText;
+      }
+      txns.push(changedTransaction);
+    }
+  }
+
+  return txns;
+}
+
+async function getAccountSummary(page) {
+  let summary;
+  try {
+    const balanceElm = await page.$('.current_balance');
+    const balance = await balanceElm.getProperty('innerText');
+    const balanceValue = getAmountData(await balance.jsonValue(), true);
+    // TODO: Find the credit field in bank website (could see it in my account)
+    summary = {
+      balance: Number.isNaN(balanceValue.amount) ? 0 : balanceValue.amount,
+      creditLimit: 0.0,
+      creditUtilization: 0.0,
+      balanceCurrency: balanceValue.currency,
+    };
+  } catch (err) {
+    // Just in case
+    summary = {
+      balance: 0.0,
+      creditLimit: 0.0,
+      creditUtilization: 0.0,
+      balanceCurrency: 'ILS',
+    };
+  }
+  return summary;
+}
+
+async function fetchTransactionsForAccount(page, startDate) {
+  const summary = await getAccountSummary(page);
+  await waitUntilElementFound(page, 'input#fromDate');
+  // Get account number
+  const snifNmbr = await page.$eval('.branch_num', (span) => {
+    return span.innerText;
+  });
+
+  const accountNmbr = await page.$eval('.acc_num', (span) => {
+    return span.innerText;
+  });
+  const accountNumber = `14-${snifNmbr}-${accountNmbr}`;
+  // Search for relavant transaction from startDate
+  await clickButton(page, '#tabHeader4');
+  await fillInput(
+    page,
+    'input#fromDate',
+    startDate.format('DD/MM/YYYY'),
+  );
+
+  await clickButton(page, '#fibi_tab_dates .fibi_btn:nth-child(2)');
+  await waitForNavigation(page);
+  await waitUntilElementFound(page, 'table#dataTable077, #NO_DATA077');
+  let hasNextPage = true;
+  let txns = [];
+
+  const noTransactionElm = await page.$('#NO_DATA077');
+  if (noTransactionElm == null) {
+    // Scape transactions (this maybe spanned on multiple pages)
+    while (hasNextPage) {
+      const pageTxns = await readPage(page);
+      txns = txns.concat(pageTxns);
+      const button = await page.$('#Npage');
+      hasNextPage = false;
+      if (button != null) {
+        hasNextPage = true;
+      }
+      if (hasNextPage) {
+        await clickButton(page, '#Npage');
+        await waitForNavigation(page);
+        await waitUntilElementFound(page, 'table#dataTable077');
+      }
+    }
+  }
+
+  return {
+    accountNumber,
+    summary,
+    txns: convertTransactions(txns.slice(1)), // Remove first line which is "opening balance"
```

Comment:
@matanelgabsi this slicing could be avoided if you would parse line by line. @esakal your fault again 😄